Create a "Bowling" class, for validate input data and calcule the points of the game

jorge27 opened the pull request:
This class is a center of the game, this class validate the input when register a bowling and calcule the points of game for show in display

jorge27 committed:
Created a tests for class bowling

The class bowling most have two methods, capture and calculate.
The method capture receive two values and validate if this values are
a numbers valids for append a punctuation
The method calculate, will calcule the punctuation of the shot

jorge27 committed:
Created class "bowling", the class for validate data and calcule points

This principal methods are two, the method capture validate bowling
data points, and method calculate, calcule the points of game. Also
contains method turn, this, return the turn of the game, and method rounds,
who return the bowling points of the turn indicate

jorge27 committed:
Correct identation, and changes some spec values

Correct identation, now are 2 spaces, change some spec values, now
instead of 0, spec nil

jorge27 committed:
Create file for configure enviroment

Create Gemfile and gitignore file for configurate my enviroment

jorge27 committed:
Create main.rb, this file use Bowling class and execute the game

This file link all files and create the flow for the game.
In this case, here capture data input for call the class and later
validate and calculate the punctuation.

jorge27 committed:
Move file to app folder for use rubycritic

jorge27 committed:
Move file bowling.rb to folder app for execute rubycritic

jorge27 committed:
Modified main.rb, split the class Bowling in Bowling and Score

Modified main for append score class, and append the call to method
bowPrint, now, bowPrint is called from score class

jorge27 committed:
Split the class "Bowling" in two, "Bowling" and "Score"

Now, the class "Score" is on charge the print the punctuation and,
bowling class, capture and validate data input

ese-varo commented on app/main.rb:
```diff
@@ -0,0 +1,33 @@
+require_relative "lib/bowling"
```
Normalmente las comillas dobles se usan solo para interpolacion `"hey #{name}"`, hacer escape de algun caracter `print "\n"` o intercaladas con comillas simples `"I'm a Brightcoder"`, si no requieres alguna de esas dos opciones lo mejor es usar commitllas simples, veo que tienes el mismo detalle en varias sentencias de tu codigo

ese-varo commented on app/lib/score.rb:
```diff
@@ -0,0 +1,38 @@
+  end
+
+  def bowling_print_lines(line = '-')
+    (0..130).each {print line}
```
tienes mas de una linea donde utilizas `each` para imprimir un numero especifico de caracteres, utilizar un each para este tipo de situacion e iterar sobre un rango `(0..130)` no es lo mas optimo, a menos que quieras manipular tu print y para escribir algo mas que solo el mismo caracter, para este caso hay varias opciones, se me ocurren dos un poco mas simples:
- un `times` esta es un poco parecida a lo que ya tienes pero creo que es mas semantica: 
`130.times { print line }`
- lo mas sencillo seria imprimir tu caracter multiplicado por el numero de veces que lo necesitas: 
`print line * 130`

jorge27 committed:
Modified print score method, now for print board call score.bow_print

Now, for call the method who print the score board used bow_print in
class score

jorge27 committed:
Create private classes print_lines and print_spaces

This classes help to print the score board

jorge27 committed:
Delete getters, now, declare class propieties with attr_reader

Also defined some methods as private methods, and modified some if's for
show in one line
